tools/cpu_graph_visualizer/visualizer.py

```diff
@@ -0,0 +1,586 @@
+            output_src = svg
+        htmlfile = open(filename,'w')
+        htmlfile.write(output_src)
+        htmlfile.close()
+        return
+    return graph_src, data_map
+
+def serialize_model(self, model_path):
+    weight_path = model_path[:model_path.find(".xml")] + ".bin"
+    pass_manager = Manager()
+    pass_manager.register_pass("Serialize", model_path, weight_path)
+    pass_manager.run_passes(self)
+    return model_path, weight_path
+
+ov.Model.serialize = serialize_model
+ov.Model.print = print_model
+ov.Model.visualize = visualize_model
+
+def fill_tensors_with_random(input, shape):
+    dtype = get_dtype(input.get_element_type())
+    rand_min, rand_max = (0, 1) if dtype == np.bool else (np.iinfo(np.uint8).min, np.iinfo(np.uint8).max)
+    # np.random.uniform excludes high: add 1 to have it generated
+    if np.dtype(dtype).kind in ['i', 'u', 'b']:
+        rand_max += 1
+    rs = np.random.RandomState(np.random.MT19937(np.random.SeedSequence(0)))
+    a = rs.uniform(rand_min, rand_max, list(shape)).astype(dtype)
+    return ov.Tensor(a)
+
+def test_infer_queue(compiled_model, input_shapes, num_request, num_infer, time_limit=60):
+    infer_queue = ov.AsyncInferQueue(compiled_model, num_request)
+
+    latency_list = []
+    prof_list = []
+    def callback(request, userdata):
+        latency_list.append(request.latency)
+        prof_list.append(request.profiling_info)
+
+    infer_queue.set_callback(callback)
+
+    all_input = {}
+    for port, input in enumerate(compiled_model.inputs):
+        if port < len(input_shapes):
+            static_shape = input_shapes[port]
+        else:
+            static_shape = input.get_shape()
+        all_input[port] = fill_tensors_with_random(input, static_shape)
+
+    for i in range(num_request):
+        infer_queue.start_async(all_input, userdata=i)
+
+    t0 = time.time()
+    for i in range(num_infer):
+        wtime = time.time() - t0
+        if time_limit and (wtime > time_limit):
+            break
+        infer_queue.start_async(None, userdata=i)
+    infer_queue.wait_all()
+    fps = i/wtime
+    return latency_list, prof_list, fps, wtime
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser("CPU graph visualizer")
+    parser.add_argument("model", type=str, help="Model file path")
+    parser.add_argument("-p","--perf", action="store_true", help="do profiling")
+    parser.add_argument("--raw", action="store_true", help="also dump raw ngraph model")
+    parser.add_argument("--bf16", action="store_true", help="enable inference precision with bf16")
+    parser.add_argument("--nthreads", type=int, default=4, help="set INFERENCE_NUM_THREADS for profiling")
+    parser.add_argument("-r","--reshape", type=str, default="()", help="reshape before visualize/profiling")
+    parser.add_argument("-i","--input_shapes", type=str, default="()",
+                        help="python tuple/list of static shapes used for profiling, e.g. [[6,23,240],[5,10,2]]")
```

Comment:
```suggestion
    parser.add_argument("model", type=str, help="...")
    parser.add_argument("-p","--perf", action="store_true", help="...")
    parser.add_argument("--raw", action="store_true", help="...")
    parser.add_argument("--bf16", action="store_true", help="...")
    parser.add_argument("--nthreads", type=int, default=4, help="Sets INFERENCE_NUM_THREADS hint for profiling.")
    parser.add_argument("-r","--reshape", type=str, default="()", help="Reshape model before visualize/profiling.")
    parser.add_argument("-i","--input_shapes", type=str, default="()",
                        help="Tuple or List of static shapes used for profiling, e.g. (6,10,256) or [[6,23,240],[5,10,2]]")
```

Please align `...` it with other comment about usage. I do not see mentioning of threads, reshape and inputs as well. I am also not sure about default value for threads, it might differ between devices and machines.

Reply:
I see... So if this setting is set to 4 and run on machine with only 2 threads -- what is the result of that action? It might be a good thing to notify users about change going under the hood or just add note about it in help string.

Reply:
OK, I changed help string to indicate default value of 4 for --nthreads, also the tool will show the full device properties/configurations in `Compiling model for {} device with properties: ` part, so user can easily check out all properties used for profiling